src/index.js

```diff
@@ -52,55 +53,56 @@ export default function compile(services: Services, blocks: Blocks): GraphResult
 
   Object.keys(services).forEach((key) => {
     const {
+      id,
       name,
       execution,
       mapping,
     }: ServiceShape = services[key];
 
-    nodes.add(name);
+    nodes.add(id);
 
     const mappings: Mapping = mapping ?
-      mapping.reduce((rest, { name: alias, mapping: block }) => ({
+      mapping.reduce((rest, { id, mapping: block }) => ({
         ...rest,
-        [alias]: block,
+        [id]: block,
       }), {}) : {};
 
-    const resolveName = alias => (mappings[alias] || alias);
+    const resolveId = id => (mappings[id] || id);
```

Comment:
Same for this line. Not sure why because we never change the value of `id` it looks like...